src/Filesystem/Provider/FileStreamAccess.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+namespace ILIAS\Filesystem\Provider;
+
+use ILIAS\Filesystem\Exception\FileAlreadyExistsException;
+use ILIAS\Filesystem\Exception\FileNotFoundException;
+use ILIAS\Filesystem\Exception\IOException;
+use Psr\Http\Message\StreamInterface;
```

Comment:
I just saw that you kind of had the same question. I would prefer our own stream interface over using that one, see RAII.